code/parse/sexp.cpp

```diff
@@ -19804,6 +19805,36 @@ void multi_sexp_set_respawns()
 	}
 }
 
+/**
+ * set a hotkey for one or more ships/wings
+ */
+void sexp_set_hotkey(int node)
+{
+	int objnum, setnum, n = node;
+	setnum = eval_num(n);
+	n   = CDR(n);
+
+	// first, look for ship name -- if found, then set hotkey to ship
+	// if not a ship, look for wing name, then set hotkey to each ship in wing
+	// note, hud_target_hotkey_add_remove() checks if the object has arrived or is destroyed or departed
+	for (; n != -1; n = CDR(n)) {
+		object_ship_wing_point_team oswpt;
+		sexp_get_object_ship_wing_point_team(&oswpt, CTEXT(n));
+		if (oswpt.type == OSWPT_TYPE_SHIP) {
+			objnum = oswpt.shipp->objnum;
+			if (objnum != -1)
+				hud_target_hotkey_add_remove(setnum, &Objects[objnum], HOTKEY_MISSION_FILE_ADDED);
+		}
+		if (oswpt.type == OSWPT_TYPE_WING) {
```

Comment:
You could make this an `else if`.  Minor nitpick.

Reply:
Easy adjustment, though!